openstack/compute/v2/extensions/attachinterfaces/requests.go

```diff
@@ -11,3 +11,54 @@ func List(client *gophercloud.ServiceClient, serverID string) pagination.Pager {
 		return InterfacePage{pagination.SinglePageBase(r)}
 	})
 }
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToAttachInterfacesCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts specifies parameters of a new interface attachment.
+type CreateOpts struct {
+
+	// PortID is the ID of the port for which you want to create an interface.
+	// The NetworkID and PortID parameters are mutually exclusive.
+	// If you do not specify the PortID parameter, the OpenStack Networking API
+	// v2.0 allocates a port and creates an interface for it on the network.
+	PortID string `json:"port_id,omitempty"`
+
+	// NetworkID is the ID of the network for which you want to create an interface.
+	// The NetworkID and PortID parameters are mutually exclusive.
+	// If you do not specify the NetworkID parameter, the OpenStack Networking
+	// API v2.0 uses the network information cache that is associated with the instance.
+	NetworkID string `json:"net_id,omitempty"`
+
+	// Slice of FixedIPs. If you request a specific FixedIP address without a
+	// NetworkID, the request returns a Bad Request (400) response code.
+	FixedIPs []FixedIP `json:"fixed_ips,omitempty"`
+
+	// A device role tag that can be applied to a network interface when
+	// attaching it to the VM.
+	// The guest OS of a server that has devices tagged in this manner can access
+	// hardware metadata about the tagged devices from the metadata API and on
+	// the config drive, if enabled.
+	Tag string `json:"tag,omitempty"`
```

Comment:
It looks like `tag` was added in [microversion 2.49](https://github.com/openstack/nova/blob/stable/pike/nova/api/openstack/compute/schemas/attach_interfaces.py#L50-L52). We still haven't determined an official way of implementing features introduced in microversions. Because of that, let's remove `Tag` for now.